src/shared/hostname-setup.c

```diff
@@ -46,7 +46,7 @@ bool get_hostname_filtered(char ret[static HOST_NAME_MAX + 1]) {
 
         /* Returns true if we got a good hostname, false otherwise. */
 
-        if (gethostname(buf, sizeof(buf) - 1) < 0)
```

Comment:
i wonder why we have get_hostname_filtered() at all, if we have gethostname_strict() already? @keszybz?

Reply:
It allows "localhost" or friends. Now `get_hostname_filtered()` is replaced with `gethostname_full()`.